R/pre_commit.R

```diff
@@ -0,0 +1,220 @@
+
+  # Read tracking files
+  tryCatch(
+    {
+      log_files <- utils::read.csv(datafile_log, stringsAsFactors = FALSE)
+      ign_files <- utils::read.csv(ignore_file,
+        header = FALSE,
+        stringsAsFactors = FALSE, col.names = "filename"
+      )
+    },
+    error = function(e) {
+      message("Error reading configuration files: ", e$message)
+      return(FALSE)
+    }
+  )
+
+  message("=== Gitignore Validation ===")
+  if (ncol(ign_files) > 1) {
+    message("ERROR: Commas detected in .gitignore")
+    all_ok <- FALSE
+  }
+
+  space_files <- ign_files$filename[grepl(" ", ign_files$filename)]
+  if (length(space_files) > 0) {
+    message(
+      "ERROR: Spaces in .gitignore entries:\n",
+      paste("-", space_files, collapse = "\n")
+    )
+    all_ok <- FALSE
+  }
+
+  message("\n=== Data File Validation ===")
+  current_files <- data.frame(
+    files = list.files("./", recursive = TRUE, full.names = TRUE)
+  ) |>
+    dplyr::filter(
+      grepl(suffixes, files, ignore.case = TRUE),
+      !grepl("renv|datafiles_log.csv", files)
+    )
+
+  for (file in current_files$files) {
+    rel_path <- gsub("^./", "", file)
+
+    if (!rel_path %in% log_files$filename) {
+      message(
+        "Missing entry: ", rel_path, "\n",
+        "  - Add to datafiles_log.csv with status: ",
+        "published, reference, or dummy\n"
+      )
+      all_ok <- FALSE
+      next
+    }
+
+    status <- tolower(log_files$status[log_files$filename == rel_path])
+    in_ignore <- rel_path %in% ign_files$filename
+
+    if (!status %in% valid_statuses) {
+      if (!in_ignore && !grepl("unpublished", rel_path)) {
+        message(
+          "Unmanaged file: ", rel_path, "\n",
+          "  - Status: ", log_files$status[log_files$filename == rel_path], "\n",
+          "  - Valid statuses: ", paste(valid_statuses, collapse = ", "), "\n",
+          "  - Add to .gitignore if unpublished\n"
+        )
+        all_ok <- FALSE
+      } else {
+        message(
+          "Note: ", rel_path,
+          " is unpublished (ignored) but consider updating its status\n"
+        )
+      }
+    }
+  }
+
+  if (all_ok) {
+    message("\nAll data file checks passed!")
+    return(TRUE)
+  }
+
+  message("\nCommit blocked: Data validation issues detected")
+  return(FALSE)
+}
+
+#' Check for and Replace Google Analytics Keys
+#'
+#' Reads `google-analytics.html` and `ui.R` to check for Google Analytics keys.
+#' If found, replaces them with placeholder values and stages the files in git.
+#'
+#' @param ga_file Path to the Google Analytics file.
+#' @param ui_file Path to the UI script file.
+#' @return TRUE if no issues found, FALSE if replacements were made.
+#' @importFrom xfun gsub_file
+#' @export
+check_analytics_key <- function(ga_file = "google-analytics.html",
+                                ui_file = "ui.R") {
+  ga_pattern <- "(G-[A-Z0-9]{10}|UA-\\d{6,9}-\\d+)"
+
+  ga_content <- readLines(ga_file)
+  ui_content <- readLines(ui_file)
```

Comment:
Looks like you didn't use ui_content within the script in the end, so should just remove this line